app/models/board.rb

```diff
@@ -10,34 +10,27 @@ class Board < ApplicationRecord
   belongs_to :creator, class_name: 'User', inverse_of: false, optional: false
 
   has_many :board_authors, inverse_of: :board, dependent: :destroy
+  has_many :authors, class_name: 'User', through: :board_authors, source: :user, dependent: :destroy
   has_many :board_coauthors, -> { where(cameo: false) }, class_name: 'BoardAuthor', inverse_of: :board
   has_many :coauthors, class_name: 'User', through: :board_coauthors, source: :user, dependent: :destroy
   has_many :board_cameos, -> { where(cameo: true) }, class_name: 'BoardAuthor', inverse_of: :board
   has_many :cameos, class_name: 'User', through: :board_cameos, source: :user, dependent: :destroy
 
   validates :name, presence: true, uniqueness: true
 
+  after_create :add_creator_to_authors
   after_destroy :move_posts_to_sandbox
 
   scope :ordered, -> { order(pinned: :desc, name: :asc) }
 
-  def writers
-    @writers ||= coauthors + [creator]
-  end
-
-  def writer_ids
-    coauthor_ids + [creator_id]
-  end
-
   def open_to?(user)
     return false unless user
-    return true if open_to_anyone?
-    return true if creator_id == user.id
+    return true unless self.authors_locked?
```

Comment:
I'd keep the creator check here; it's logically unnecessary but it does save on the db call.